pkg/controllers/plugin/util.go

```diff
@@ -53,3 +62,127 @@ func convertRuntimeObjectToCoreV1Service(o interface{}) (*corev1.Service, error)
+		// Check if all pods are running, if one of them is not running, return false
+		for _, pod := range obj.Items {
+			if pod.Status.Phase != corev1.PodRunning {
+				return false
+			}
+		}
+		return true
+	}
+	return false
+}
+
+func fetchPodList(labelSelector map[string]string, nameSpace string, cl client.Client) *[]PodStatus {
+	var podList = new(corev1.PodList)
+	var podStatusList = new([]PodStatus)
+	listOptions := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labelSelector),
+		Namespace:     nameSpace,
+	}
+	if err := cl.List(context.Background(), podList, listOptions); err != nil {
+		return nil
+	}
+
+	for _, pod := range podList.Items {
+		podStatus := PodStatus{
+			Name:       pod.Name,
+			NodeName:   pod.Spec.NodeName,
+			Conditions: checkPodStatusCondition(pod.Status.Conditions),
+		}
+		switch pod.Status.Phase {
+		case corev1.PodRunning:
+			podStatus.Phase = true
+		case corev1.PodSucceeded:
+			podStatus.Phase = true
+		default:
+			podStatus.Phase = false
+		}
+		*podStatusList = append(*podStatusList, podStatus)
+	}
+	return podStatusList
+}
+
+func checkPodStatusCondition(podCondition []corev1.PodCondition) bool {
+	for _, condition := range podCondition {
+		switch condition.Type {
+		case corev1.PodInitialized:
+			if condition.Status != corev1.ConditionTrue {
+				return false
+			}
+		case corev1.PodReady:
+			if condition.Status != corev1.ConditionTrue {
+				return false
+			}
+		case corev1.ContainersReady:
+			if condition.Status != corev1.ConditionTrue {
+				return false
+			}
+		case corev1.PodScheduled:
+			if condition.Status != corev1.ConditionTrue {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func composeMessage(podStatusList *[]PodStatus) string {
+	var message string
+	for _, pod := range *podStatusList {
+		if !pod.Conditions {
+			message = message + ",Pod Name: " + pod.Name + " on Node: " + pod.NodeName + "\n"
+		}
+	}
+	return message
+}
+
+func computeReadyCondition(pluginStatus greenhousev1alpha1.PluginStatus, release *ReleaseStatus) greenhousev1alpha1.Condition {
+	WorkloadReadyStatus := *pluginStatus.GetConditionByType(greenhousev1alpha1.WorkloadReadyCondition)
+
+	if release.ReleaseOK {
+		WorkloadReadyStatus.Status = metav1.ConditionTrue
+		WorkloadReadyStatus.Message = "Workload is running"
+	} else {
+		WorkloadReadyStatus.Status = metav1.ConditionFalse
+		WorkloadReadyStatus.Message = release.Message
+	}
+	return WorkloadReadyStatus
+}
+
+func logPluginStatus(ctx context.Context, releaseStatus *ReleaseStatus) {
+	marshaled, err := json.Marshal(releaseStatus)
+	if err != nil {
+		log.FromContext(ctx).Info("releaseStatus marshaling error", "error", err)
+	}
+	if releaseStatus.ReleaseOK {
+		log.FromContext(ctx).Info("release is running", "pluginName", releaseStatus.ReleaseName, "Details", string(marshaled))
+	} else {
+		log.FromContext(ctx).Info("release is not running", "pluginName", releaseStatus.ReleaseName, "Details", string(marshaled))
+	}
+}
```

Comment:
I like the idea of providing more information w.r.t. the release's workload resources.
WDYT about exposing a metric instead e.g. 
```
greenhouse_plugin_workload_status{org="ccloud", pluginDefinitIon="alerts", plugin="alerts", status="?"}
```
Then we can also alert if we see that workload deployed by Greenhouse starts to fail 💭

Reply:
:100: